modules/backend/formwidgets/ColorPicker.php

```diff
@@ -78,10 +80,37 @@ public function prepareVars()
     {
         $this->vars['name'] = $this->getFieldName();
         $this->vars['value'] = $value = $this->getLoadValue();
-        $this->vars['availableColors'] = $this->availableColors;
+        $this->vars['availableColors'] = $this->getAvailableColors();
         $this->vars['allowEmpty'] = $this->allowEmpty;
         $this->vars['showAlpha'] = $this->showAlpha;
-        $this->vars['isCustomColor'] = !in_array($value, $this->availableColors);
+        $this->vars['isCustomColor'] = !in_array($value, $this->vars['availableColors']);
+    }
+
+    /**
+     * Gets the appropriate list of colors.
+     *
+     * @return array
+     */
+    protected function getAvailableColors()
+    {
+        if (is_array($this->availableColors)) {
+            return $this->availableColors;
+        }
+        elseif (is_string($this->availableColors) && !empty($this->availableColors)) {
+            if ($this->model->methodExists($this->availableColors)) {
+                return $this->model->{$this->availableColors}(
```

Comment:
Change to `return $this->availableColors = $this->model->{$this->availableColors}` to locally cache the results of that method and also put the elseif on the same line as the end bracket please.